Adjust result sorter to work across keys

Buckwich opened the pull request:
### Proposed Changes

<!--

Add relevant context (issue fixed or related to), 
a capture of the UI changes (if any) as well as 
steps to try out your changes.

--> 

### Checklist

To ensure you provided everything we need to look at your PR:

* [ ] **Brief textual description** of the changes present
* [ ] **Visual demo** attached
* [ ] **Steps to try out** present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [ ] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

<!--

Thanks for creating this pull request! ❤️

-->

nikku commented on lib/features/popup-menu/PopupMenuComponent.js:
```diff
@@ -88,8 +88,8 @@ export default function PopupMenuComponent(props) {
         'label',
-        'description',
-        'search'
+        'search',
```
:star:

nikku commented:
As per https://github.com/camunda/camunda-modeler/issues/4856#issuecomment-2754045222, let's ensure we can test this isolated in diagram-js, in a big enough, close to real work example.

nikku reviewed (changes requested):
Todo to get this merged:

* Test to assert priority (higher up matches beat lower ones)
* Clean up test fixtures to only contain relevant fields (https://github.com/bpmn-io/diagram-js/pull/963/files#diff-1f6fe72e724cb08e8d1db708b6b332f7b5c6ce9cb40686e09023fa4ea74682f0)
* Consider if the real world test should be in search items, or `PopupMenuSpec`. I'd rather have it in popup menu spec to verify the behavior E2E. 
* Name the `searchitems.json` file to be grouped with the spec (cf.  [existing pattern](https://github.com/bpmn-io/bpmn-js/tree/develop/test/spec)). When part of `PopupMenu.spec`, call it `PopupMenu.complex-items.json` or the like.

Buckwich commented:
I've squashed commits you previously reviewed and dropped the real world example from this branch. I'll rework that properly with your suggestions in #966. also adjusted the test to be simpler and concrete

nikku commented on test/spec/features/search/searchSpec.js:
```diff
@@ -144,13 +144,72 @@ describe('features/search', function() {
+    ];
+    const searchItems = (items, term) => search(items, term, {
+      keys: [
+        'search',
```
What is confusing here is that we switch keys compared to previous setups. What if we (1) keep keys and items or (2) swap items (input) instead?

Buckwich replied:
original idea was to make it clear that i want to test the different order, i was concerned otherwise it could seem that it just takes the original order. i hope i made i more clear now, I think I applied you suggestion

nikku commented on test/spec/features/search/searchSpec.js:
```diff
@@ -144,13 +144,72 @@ describe('features/search', function() {
   }));
 
 
+  it('should prioritize start of search', inject(function(search) {
```
```suggestion
  it('should prioritize secondary key start of word', inject(function(search) {
```

nikku commented on test/spec/features/search/searchSpec.js:
```diff
@@ -144,13 +144,72 @@ describe('features/search', function() {
+      {
+        label: 'User task',
+        search: 'form human',
+      } ];
```
```suggestion
      }
    ];
```

nikku replied:
(sorry for that one :see_no_evil:)

nikku reviewed (approved):
Looks good to me, see [this test](https://github.com/bpmn-io/diagram-js/pull/963/files#diff-7dc68f128643d145cf2a06ab4ce54a7cc10388034805b1f6cbe730303853583dR1498) verifying that popup menu works according to your changes.

Updated your changes to be a `feat`. I think it is not a bug fix, but an improvement of the existing search.

Buckwich commented:
> Updated your changes to be a feat. I think it is not a bug fix, but an improvement of the existing search.

Thanks. I originally choose fix as this was in response to a bug report
